### PR TITLE
misc: fixes and cleanups found during multi-testing

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Split misc fixes and cleanups from https://github.com/pmodels/mpich/pull/5725.
Fixes in `ch4//release_gather`, `maint/extractcvars`, `test/mpi/coll/op_coll.c`, `test//coll_cvars.txt`, and in testlist syntax.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/shm/posix/posix_coll_release_gather.h [hunk 1]
```diff
@@ -53,7 +53,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_bcast_release_gather(void *buffer,
     MPI_Aint lb, true_lb, true_extent, extent, type_size;
     void *ori_buffer = buffer;
     MPI_Datatype ori_datatype = datatype;
-    int cellsize = MPIDI_POSIX_RELEASE_GATHER_BCAST_CELLSIZE;
```
Comment (raffenet) [inline src/mpid/ch4/shm/posix/posix_coll_release_gather.h:56]: Is moving this setting later necessary? Unclear to me how this affects the behavior.

**Review by raffenet — COMMENTED** [review]
Looks good to me. Just one minor question.

File: src/mpid/ch4/shm/posix/posix_coll_release_gather.h (hunk 1 above)
Comment (hzhou) [reply src/mpid/ch4/shm/posix/posix_coll_release_gather.h:56]: It is not "necessary", but for readability. `cellsize` is only used in `MPIR_Algo_calculate_pipeline_chunk_info` and the reason we need this variable is due to the exception under `#ifdef HAVE_ERROR_CHECKING`. So move them closer makes the code easier to read. I can split this change if you prefer.

Comment (hzhou) [reply src/mpid/ch4/shm/posix/posix_coll_release_gather.h:56]: I have split the commits.

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Review by raffenet — APPROVED** [review]

**Pull request merged**